libparsec/crates/tests_macros/src/lib.rs

```diff
@@ -97,6 +86,57 @@ pub fn parsec_test(attr: TokenStream, item: TokenStream) -> TokenStream {
     })
 }
 
+fn validate_testbed_argument(sig: &mut syn::Signature) -> ControlFlow<TokenStream> {
+    match sig.inputs.pop().map(Pair::into_value) {
+        Some(FnArg::Typed(typed)) => {
+            validate_testbed_argument_name(&typed)?;
+            validate_testbed_argument_type(&typed)?;
+        }
+        _ => {
+            return ControlFlow::Break(compile_error(
```

Comment:
The function `compile_error` provide better error message.

Instead of having the message `your macro has panic'ed` with the actual error message somewhere, that function make the actual error message to be the error message and provide the `span` (the piece of code that produced the message)

Reply:
I've renamed it to `generate_compile_error`